### DISCUSSION TITLE
Personal ('pet name') room name for chats

### DISCUSSION THREAD

**Discussion by dreamflasher** [element-hq/element-meta#726] [Ideas]
Currently, if the channel with another person is not named, then their nickname is shown. 
Problem: 
1) It's impossible with many people to remember everyones nickname.
2) Also it's impossible to convince everyone to use their full real name (and also undesired).
3) Naming the channel in a meaningful way has the problem that it's shown for the other person, too. So currently we use titles for chats where both names are in. 

Second group of problems: A MUC rarely has a good name that is helpful to everyone. See @uhoreg comment. Eg. you have an "Organization" channel of some open source project, which is just very unhelpful if you are in multiple projects. So it would in general be helpful for all chats to have a custom/personal room name.

Possible solutions:
- Make the channel name for People chats a personal/local/account setting (instead of a server? setting)
- Have something like an address book where one can give names for people and display these names when the channel is unnamed

**Comment by uhoreg** [comment]
It would be helpful for users to be able to set a custom name for rooms in general, with the custom name only being shown to the user who set it.  For example, it would be useful if someone is subscribed to both #debian on Freenode and #debian on OFTC to set custom names so that they're distinguishable.

**Comment by dreamflasher** [comment]
Good point, I didn't think of it, yes actually channel names should personalizable in general.

**Comment by ara4n** [comment]
This is basically https://github.com/matrix-org/matrix-doc/issues/690.  Agreed it'd be nice to have - it's very annoying when someone renames a 1:1 to be called something unhelpful like "Matthew"

**Comment by uhoreg** [comment]
Would also be good to be able to set a personal avatar for the room without affecting other users.

**Comment by turt2live** [comment]
Pet avatars are tracked here ftr: https://github.com/vector-im/riot-web/issues/363

**Comment by MurzNN** [comment]
For don't create new entities in storage, can we store 'pet name' for rooms in `account_data` like `ignored_users` stored now? 
Something like:
```
{
    "content": {
        "room_names": {
            "!QtykxKocfZaZOUrTwp:matrix.org": {
                "name": "Matrix flame"
            }
        }
    },
    "type": "m.room_name_override"
}
```

**Comment by t3chguy** [comment]
Why not just store it in room account data then you don't need the room id inside the event.

**Comment by MurzNN** [comment]
> Why not just store it in room account data then you don't need the room id inside the event.

Does room account data have private storage for each user, that not visible to other room members?

**Comment by t3chguy** [comment]
Yup, that's why it's room *account* data.

**Comment by kiliankoe** [comment]
This would be fantastic to have. I'm one of those people who have trouble remembering usernames of my contacts. I also have groups with certain users that I would love to just label with something related to these users that doesn't make sense for them (e.g. "parents").

**Comment by eternalbit** [comment]
First of all, sorry for not adding real content to the discussion.
Moderators, feel free to delete my comment.

My point: I don't understand why this issue has **p2** assigned. I've literally seen roughly 10 more or less duplicate issues with this same request from users, who also want this feature - going back to 2017.

And then there are issues with **p1** assigned where I really don't get how it's supposed to be more important than this one here ... example: [Why doesn't 'da<tab>' autocomplete Dave in #riot-dev?](https://github.com/vector-im/riot-web/issues/8957)

Sure, priorities are somehow a matter of personal perspective ... but for me, the same issue popping up again and again like this one clearly indicates it's quite important.

And I bet that this is one of the missing features that keep potential users (like me) from using riot/vector.im. I don't want to have to remember silly pseudonyms - which is what many private users chose when making an account. Sure, users from french administration probably use their actual names anyway. But if riot/vector.im becomes more widespreach they won't be a big percentage of all users anymore.

**Comment by dukelec** [comment]
+1

**Comment by MurzNN** [comment]
Seems this is planned to implement in https://github.com/vector-im/roadmap/issues/10 roadmap item?

**Reply by flo1o1** [reply]
No it's not, at least not specifically "allow my own contact names for others, just visible to me".

**Comment by MurzNN** [comment]
I have created the https://github.com/matrix-org/matrix-doc/pull/3015 for this feature, hope this will be implemented in some clients.

**Comment by vmario89** [comment]
found that issue because i was looking for the same missing feature. happy to see if we could rename persons locally wihtout overriding global room names, to have a better overview of a whole community and a lot of group members and friends.

**Comment by SZemanek** [comment]
It would be super cool if this could integrate with contacts on the android version of element.

**Comment by ThePharaohArt** [comment]
Hopefully allowing us to give nicknames or pet names to friends on the app to better distinguish them from other friends will arrive soon.

**Comment by kittykat** [comment]
Moving this issue to discussions in Element meta as we need to make a cross platform decision on how to proceed :+1:

**Comment by dreamflasher** [comment]
6 years later... any update on this?

**Reply by eNkKVS4HxHc9jaq** [reply]
I know right? This seems to be one of the most engaged threads but no plans for implementation seem to have been made...

**Reply by dreamflasher** [reply]
I got used to it. And it makes sense: Nobody is getting paid for what users want, especially not in open source. So it's fair that open source developers work on whatever they like, it's their time. And every developer prefers developing a shiny new feature, so here we are :)

**Reply by austin2Go** [reply]
> I got used to it. And it makes sense: Nobody is getting paid for what users want, especially not in open source. So it's fair that open source developers work on whatever they like, it's their time. And every developer prefers developing a shiny new feature, so here we are :)

It seems like a no-brainer. Rather bizarre that years later, such a basic feature that users _want_ hasn't been implemented. There are some things that make no sense in this world.

**Comment by K4LCIFER** [comment]
Locally changing names is rather crucial before Element will have widespread adoption. You should be able to set your own default name for a room, your default name globally, and other people should be able to give you their own local custom name for the room.

**Comment by gitayam** [comment]
Hi hopefully keeping this thread alive as I'd really love having this feature

**Comment by tomasalves8** [comment]
+1

**Comment by everanium** [comment]
+∞

**Comment by ParaN3xus** [comment]
+1

**Comment by ollietheotter983** [comment]
+1

**Comment by gitayam** [comment]
+1

**Comment by DarkArc** [comment]
> Have something like an address book where one can give names for people and display these names when the channel is unnamed

I think this is a really important issue for mainstream usability of Matrix/Element in general.

The scope of this problem is "contact management"; it goes beyond naming a room.

## Personal Contact Management

It includes managing how you see the individual across the platform, both in things like their name _AND_ their profile picture.  Some examples of why you might want this:
- If your close friend "Bob Jones" goes by on Matrix as "Daffy Duck" and has a picture of "Daffy Duck" meanwhile, you'd prefer to see "Bob Jones" and a picture of Bob Jones the person.
- You know multiple people that go by "Daffy" and you need to clearly differentiate them.
- You want to make pictures (e.g., caricatures) for everyone you know and use those as profile pictures to provide a consistent look and feel to your chat app.
- Your friend or family member is using a very old pixelated, "outright weird", or group photo for their profile picture.

## Community Contact Management

This also goes into not just personal contact management but community contact management. Admins in a particular space or room might want to enforce a naming scheme (e.g., everyone goes by their gamer names). Discord allows this latter problem to be solved by "server names" for this exact use case.

## Poor UX of the Status Quo

The ability to "name" a 1-on-1 chat also seems like functionality that just shouldn't be a thing. The naive approach to "contact management" is to rename the "room" which is in the modern Element apps modeled to be a "person" themselves not "1-on-1 chats with people you know." Thus, it's very surprising when you rename the "Daffy Duck" room to "Bob Jones" and suddenly both you _and_ Bob Jones are looking at a chat called "Bob Jones." If Bob Jones has multiple contacts do this to him after joining Matrix/Element, Bob Jones now has 5 chats with himself; that's an **_absolutely awful_** user experience.

## Additional Use Cases/Considerations

Consider using Matrix with professional colleagues, friends known in person, friends known online, and family. In all four cases, you may want to also _present_ your profile differently. Having a way to accommodate these relationships (i.e., "social circles" of sorts) and provide a different default profile would be a good idea (e.g., you could be "Robert Jones" in a suite for the professional world, "Bob Jones" in a sports shirt to your friends, "Zezima" with a game screenshot as a profile picture in your gaming circles, and just "Bobby" with a picture in the recliner to your family ... you might also opt to have your profile picture and display name be one of these four or just totally empty for non-contacts).

This could work by having a default "profile" and then getting prompted to choose the profile to use when enter a new chat or messaging a user for the first time. For use cases where users do not want to express multiple profiles, no such prompts should of course occur.

**Reply by lbernath1** [reply]
I wholeheartedly agree. 
Just one minor remark: Because a room can be part of multiple spaces, I am not sure if there is an easy way to have space wide community contact management. 
But I think the implementation of a basic personal contact management is way more important and should not be hindered by discussions about an ideal solution that encapsulates every possible case.

**Comment by tyuichis** [comment]
Bumping this -- moving from Skype (which officially closes in May 2025) and Discord, both of which already support this. It seems to be a common UI/UX design paradigm.

**Comment by WhiteSockedDancer** [comment]
Any news on this? :)
